lib/timeline/component/item/BackgroundItem.js

```diff
@@ -105,16 +99,73 @@ BackgroundItem.prototype.redraw = function() {
     // update class
     var className = (this.data.className ? (' ' + this.data.className) : '') +
         (this.selected ? ' vis-selected' : '');
-    dom.box.className = this.baseClassName + className;
+    this.dom.box.className = this.baseClassName + className;
+  }
+}
+BackgroundItem.prototype._getDomComponentsSizes = function() {
+  // determine from css whether this box has overflow
+  this.overflow = window.getComputedStyle(this.dom.content).overflow !== 'hidden';
+  return {
+    content: {
+      width: this.dom.content.offsetWidth
+    }
+  }
+}
+BackgroundItem.prototype._updateDomComponentsSizes = function(sizes) {
+  // recalculate size
+  this.props.content.width = sizes.content.width;
+  this.height = 0; // set height zero, so this item will be ignored when stacking items
 
-    // determine from css whether this box has overflow
-    this.overflow = window.getComputedStyle(dom.content).overflow !== 'hidden';
+  this.dirty = false;
+}
 
-    // recalculate size
-    this.props.content.width = this.dom.content.offsetWidth;
-    this.height = 0; // set height zero, so this item will be ignored when stacking items
+BackgroundItem.prototype._repaintDomAdditionals = function() {
+}
 
-    this.dirty = false;
+/**
+ * Repaint the item
+ * @param {boolean} [returnQueue=false]  return the queue
+ * @return {boolean} the redraw queue if returnQueue=true
+ */
+BackgroundItem.prototype.redraw = function(returnQueue) {
+  var sizes
+  var queue = [
+    // create item DOM
+    this._createDomElement.bind(this),
+
+    // append DOM to parent DOM
+    this._appendDomElement.bind(this),
+
+    // update dirty DOM. An item is marked dirty when:
+    // - the item is not yet rendered
+    // - the item's data is changed
+    // - the item is selected/deselected
+    this._updateDirtyDomComponents.bind(this),
+
+    (function() {
+      if (this.dirty) {
+        sizes = this._getDomComponentsSizes.bind(this)();
+      }
+    }).bind(this),
```

Comment:
See remark on arrow function `() => {}` elsewhere in this review. For sequences of functions like this, it really helps.

Reply:
Oh, crap. I understand now, arrow functions are not present in `timeline` because it uses old syntax.

....but.... AFAIK the babel conversion from ES6 runs over *all code*. I may be wrong.